### DISCUSSION TITLE
[Zookeeper依赖] 目前 ZK 的作用？

### DISCUSSION THREAD

**Discussion by fuhaiq** [apache/kyuubi#3393] [Q&A] [answered]
我理解目前 Zookeeper 的作用一个是 HA，一个是存 Engine 路由信息。应该就这2个吧？

---

另外，我们在 K8S 里面，HA 直接使用 Deployment 多副本 + Service。如果把路由信息以 K8S 支持的方式存储（比如 CRD），那么在 K8S 里面 ZK 就可以去掉了，不知道社区有这方面计划么？

**Comment by pan3793** [comment, marked as answer]
Basically, yes, but we also require zookeeper as distributed lock, on creating engine instance.

We got reported that some users have their own fork of Kyuubi(not open-source) and use K8s label-related information for engine routing.

We have not put those ideas into our roadmap now, but PRs are welcome.

**Reply by fuhaiq** [reply]
谢谢
